server/mediaserver.go

```diff
@@ -820,10 +826,15 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 
 	// Check for presence and register if a fresh cxn
 	if !exists {
-		appData := (createRTMPStreamIDHandler(ctx, s, authHeaderConfig))(r.URL)
-		if appData == nil {
-			errorOut(http.StatusInternalServerError, "Could not create stream ID: url=%s", r.URL)
-			return
+		appData, err := (createRTMPStreamIDHandler(ctx, s, authHeaderConfig))(r.URL)
+		if err != nil {
+			if strings.HasPrefix(err.Error(), "Forbidden") {
```

Comment:
Consider returning a const error like below instead of distinguishing errors by checking string message.

```golang
var (
    errForbidden = errors.New("Forbidden")
)
```